src/librustc/ty/context.rs

```diff
@@ -272,9 +275,10 @@ pub struct TypeckTables<'tcx> {
 }
 
 impl<'tcx> TypeckTables<'tcx> {
-    pub fn empty() -> TypeckTables<'tcx> {
+    pub fn empty(local_id_root: DefId) -> TypeckTables<'tcx> {
         TypeckTables {
-            type_dependent_defs: NodeMap(),
+            local_id_root,
```

Comment:
The thing with `local_id_root` being sometimes invalid makes me uncomfortable. Please insert an `Option` *somewhere* (I think making `local_id_root` an option would work well).